### PR TITLE
tabs added

### PR DESCRIPTION
**Pull request by brtsmrtn**

### CODE REVIEW

File: src/components/Listing.tsx [hunk 1]
```diff
@@ -1,29 +1,260 @@
...
-      title: "Links to read",
-      items: [...items.filter((item) => !item.alreadyRead)],
-      visible: true,
+import { Tab as TabPanel, Tabs } from "@material-ui/core";
+import { makeStyles, Theme } from "@material-ui/core/styles";
+import { selectTab } from "../features/tabs";
+import { ListItem } from "../types/ListItem";
+import { COLORS } from "../utils/constants";
+
+const useStyles = makeStyles((theme: Theme) => ({
```
Comment (brtsmrtn) [inline src/components/Listing.tsx:11]: This is monstrosity and I am gonna work on it when styling the app.

File: src/components/Listing.tsx [hunk 2]
```diff
@@ -1,29 +1,260 @@
...
+    // "&$selected": {
+    //   opacity: 0,
+    // },
+    "& span.MuiTab-wrapper, & span.MuiTouchRipple-root": {
+      alignItems: "flex-end",
+    },
+  },
+}));
+
+type ItemsToTabConnectorType = {
```
Comment (brtsmrtn) [inline src/components/Listing.tsx:182]: I am quite unsatisfied with this solution but could not figure anything better at the moment. Could we have a look at this next time @erikvanek?

File: src/features/tabs.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  action: TabsActions
+): TabsState {
+  switch (action.type) {
+    case ADD_NEW_TAB: {
+      return [...state, action.tab];
+    }
+    case SELECT_TAB: {
+      const unselectedTabs = [
+        ...state.map((o) => {
+          return Object.assign({}, o, {
```
Comment (erikvanek) [inline src/features/tabs.ts]: spread operator should be enough

File: src/features/tabs.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  state = initialTabsState,
+  action: TabsActions
+): TabsState {
+  switch (action.type) {
+    case ADD_NEW_TAB: {
+      return [...state, action.tab];
+    }
+    case SELECT_TAB: {
+      const unselectedTabs = [
+        ...state.map((o) => {
```
Comment (erikvanek) [inline src/features/tabs.ts]: replace `o` with someting more meaningful :)

File: src/features/tabs.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import { configureStore } from "@reduxjs/toolkit";
+import { Tab } from "../types/Tab";
+import { Tag } from "../types/Tag";
+let tabsCounter = 1;
```
Comment (erikvanek) [inline src/features/tabs.ts]: Could we compute this from `initialTabs`?

File: src/features/items.ts [hunk 6]
```diff
@@ -114,7 +114,7 @@ export function itemsReducer(
       const { item, tag } = action;
       if (item.tags.filter((t) => t.title === tag.title).length) {
         const indexOfItem = state.findIndex((it) => it.id === item.id);
-        const indexOfTag = item.tags.findIndex((tag) => tag.id === tag.id);
+        const indexOfTag = item.tags.findIndex((t) => t.id === tag.id);
```
Comment (erikvanek) [inline src/features/items.ts]: bad change

File: src/components/Listing.tsx (hunk 2 above)
Comment (erikvanek) [reply src/components/Listing.tsx:182]: We should I guess. Already the type name induces some code smells somewhere...

File: src/components/Listing.tsx (hunk 1 above)
Comment (erikvanek) [reply src/components/Listing.tsx:11]: Why do we then have it now here when we are not doing styling properly now?

File: src/components/Listing.tsx [hunk 7]
```diff
@@ -1,29 +1,260 @@
...
+        aria-label="Vertical tabs example"
+        indicatorColor="primary"
+        className={classes.tabs}
+      >
+        {itemsToTabConnector.map((tab) => {
+          let className = classes.tab;
+          if (tab.coloredWith) {
+            className += ` chip-${tab.index - 1}`;
+          }
+          if ((tab.coloredWith && tab.items.length != 0) || tab.index < 2) {
```
Comment (erikvanek) [inline src/components/Listing.tsx]: What does this condition exactly mean?

File: src/features/items.ts (hunk 6 above)
Comment (brtsmrtn) [reply src/features/items.ts]: done

**Pull request merged**